### PR TITLE
Using types for verb conjugation patterns (#685)

### PR DESCRIPTION
**Pull request by psfinaki**
Same thing as for declension patterns recently.

### CODE REVIEW

File: tests/Core.Tests/VerbPatternDetectorTests.fs [hunk 1]
```diff
@@ -78,7 +79,10 @@ let ``Detects pattern is not čistit`` word =
 let ``Gets pattern class E`` verb pattern =
     verb
     |> getPatternClassE
-    |> equals (Some pattern)
+    |> Option.get
+    |> string
+    |> fun s -> s.ToLower()
+    |> equals pattern
```
Comment (psfinaki) [inline tests/Core.Tests/VerbPatternDetectorTests.fs:85]: I know this looks lame but `InlineData` in `Xunit.Theory` has to be constant so I did this crude adjustment for now and will soon rewrite the tests to thoroughly check each subpattern.

Comment (reflash) [reply tests/Core.Tests/VerbPatternDetectorTests.fs:85]: I think you can put this into a helper method. e.g.
```fsharp
let checkPattern classGetter verb pattern =
  verb
    |> classGetter
    |> Option.get
    |> string
    |> fun s -> s.ToLower()
    |> equals pattern

let ``Gets pattern class E`` =
  checkPattern getPatternClassE
```

Comment (psfinaki) [reply tests/Core.Tests/VerbPatternDetectorTests.fs:85]: Well, class getters have different signatures, e.g. `string -> ConjugationPatternClassE` for `getPatternClassE` and `string -> ConjugationPatternClassNE` for `getPatternClassNE`, so it cannot be generalized just like this.

But that's basically a signal for writing proper tests against every subpattern which is to be done.

Comment (reflash) [reply tests/Core.Tests/VerbPatternDetectorTests.fs:85]:
```fsharp
let checkPattern (classGetter: string -> 'a) verb (pattern: 'a) =
  verb
    |> classGetter
    |> Option.get
    |> string
    |> fun s -> s.ToLower()
    |> equals pattern
```
try this then

Comment (psfinaki) [reply tests/Core.Tests/VerbPatternDetectorTests.fs]: Well if you really insist... see updated version :D

Comment (reflash) [reply tests/Core.Tests/VerbPatternDetectorTests.fs]: Perfect! Thanks :)

**Pull request merged**